tests/testSphPoint.py

```diff
@@ -0,0 +1,615 @@
+        return coords
+
+    def testInit2ArgErrors(self):
+        """Test if 2-argument form of __init__ handles invalid input.
+        """
+        # Latitude is checked.
+        with self.assertRaises(pexEx.OutOfRangeError):
+            SphPoint(0.0*degrees, ( 90.0 + 1e-10)*degrees)
+        with self.assertRaises(pexEx.OutOfRangeError):
+            SphPoint(0.0*degrees, (-90.0 - 1e-10)*degrees)
+        SphPoint(0.0*degrees, 90.0*degrees)
+
+        # Longitude is not checked.
+        SphPoint(360.0*degrees, 45.0*degrees)
+        SphPoint(-42.0*degrees, 45.0*degrees)
+        SphPoint(391.0*degrees, 45.0*degrees)
+
+        # NaNs are checked.
+        with self.assertRaises(pexEx.InvalidParameterError):
+            SphPoint(float("nan")*degrees, 45.0*degrees)
+        with self.assertRaises(pexEx.InvalidParameterError):
+            SphPoint(-42.0*degrees, float("nan")*degrees)
+
+    def testInit1ArgErrors(self):
+        """Test if 1-argument form of __init__ handles invalid input.
+        """
+        # Only one singularity, at zero
+        with self.assertRaises(pexEx.InvalidParameterError):
+            SphPoint(afwGeom.Point3D(0.0, 0.0, 0.0))
+        SphPoint(afwGeom.Point3D( 0.0,   -0.2,   0.0 ))
+        SphPoint(afwGeom.Point3D( 0.0,    0.0,   1.0 ))
+        SphPoint(afwGeom.Point3D(42.78, -46.29, 38.27))
+
+        # NaNs are checked.
+        with self.assertRaises(pexEx.InvalidParameterError):
+            SphPoint(afwGeom.Point3D(float("nan"), -0.2, 0.0))
+        with self.assertRaises(pexEx.InvalidParameterError):
+            SphPoint(afwGeom.Point3D( 0.0, float("nan"), 0.0))
+        with self.assertRaises(pexEx.InvalidParameterError):
+            SphPoint(afwGeom.Point3D( 0.0, -0.2, float("nan")))
+
+    def testInitNArgFail(self):
+        """Tests if only 1- or 2-argument initializers are allowed.
+        """
+        with self.assertRaises(NotImplementedError):
+            SphPoint()
+        with self.assertRaises(NotImplementedError):
+            SphPoint("Rotund", "Bovine")
+        with self.assertRaises(NotImplementedError):
+            SphPoint(42)
+        with self.assertRaises(NotImplementedError):
+            SphPoint("ICRS", 34.0, -56.0)
+
+    def testGetLongitudeValue(self):
+        """Test if getLongitude() returns the expected value.
+        """
+        for lon, lat in self._dataset:
+            point = SphPoint(lon*degrees, lat*degrees)
+            self.assertIsInstance(point.getLongitude(), afwGeom.Angle)
+            self.assertGreaterEqual(point.getLongitude().asDegrees(), 0.0)
+            self.assertLess(point.getLongitude().asDegrees(), 360.0)
+            if lon >= 0.0 and lon < 360.0 and abs(lat) != 90.0:
+                self.assertAlmostEqual(lon, point.getLongitude().asDegrees())
+
+    def testGetLongitudeWrap(self):
+        """Test if getLongitude() returns properly wrapped numbers.
+        """
```

Comment:
When the wrapping happens is an implementation detail, and subject to change. The spec only says that getLongitude() must return wrapped values, and it's the externally visible behavior of the class that's being tested.